packages/amplify_datastore/test/testData/Post.dart

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+import 'package:meta/meta.dart';
+
+class Post extends Model {
+  static final QueryField ID = new QueryField("id");
+  static final QueryField TITLE = new QueryField("title");
+  static final QueryField RATING = new QueryField("rating");
+  static final QueryField CREATED = new QueryField("created");
+
+  @override
+  @ModelField(targetType: 'ID', isRequired: true)
+  final String id;
+
+  @ModelField(targetType: 'String', isRequired: true)
+  final String title;
+
+  @ModelField(targetType: 'Integer', isRequired: false)
+  final int rating;
+
+  @ModelField(targetType: 'Date', isRequired: false)
+  final DateTime created;
+
+  @override
+  String getId() {
+    return id;
+  }
+
+  // Constructors
+  const Post._internal(
+      {@required this.id, @required this.title, this.rating, this.created});
+
+  factory Post(
+      {String id, @required String title, int rating, DateTime created}) {
+    return Post._internal(
+        id: id, title: title, rating: rating, created: created);
+  }

```

Comment:
Missing id creation/validation. We should probably update these models to the final ones we expect from codegen in a followup PR.

Reply:
Not needed in this PR